### PR TITLE
xtypes assert

### PR DESCRIPTION
**Pull request by LuisGP**
This PR includes:
- xtypes_assert implementation.
- Port of already defined assertions, adding a message.

### CODE REVIEW

File: include/xtypes/Assert.hpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+#define xtypes_assert2_(cond, msg) xtypes_assert3_(cond, msg, false)
+
+#define xtypes_assert3_(cond, msg, bt)                                                                              \
+    {                                                                                                               \
+        if (!(cond))                                                                                                \
+        {                                                                                                           \
+            std::string msg__ = "[XTYPES]: ";                                                                       \
+            msg__.append(__FILE__ + std::string(":") + std::to_string(__LINE__) + std::string(" - "));              \
+            msg__.append(std::string("Assertion failed with message: "));                                           \
```
Comment (lemunozm) [inline include/xtypes/Assert.hpp]: Why not a stringstream?

File: include/xtypes/Assert.hpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+#if !defined(NDEBUG)
+
+#define xtypes_assert2_(cond, msg) xtypes_assert3_(cond, msg, false)
+
+#define xtypes_assert3_(cond, msg, bt)                                                                              \
+    {                                                                                                               \
+        if (!(cond))                                                                                                \
+        {                                                                                                           \
+            std::string msg__ = "[XTYPES]: ";                                                                       \
```
Comment (lemunozm) [inline include/xtypes/Assert.hpp]: I like the `__` at the end of the variable name :+1:

File: include/xtypes/Assert.hpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            std::abort();                                                                                           \
+        }                                                                                                           \
+    }                                                                                                               \
+
+#else
+#define xtypes_assert2_(cond, msg)
+#define xtypes_assert3_(cond, msg, bt)
+#endif
+
+#define GET_MACRO(_1, _2, _3, NAME, ...) NAME
```
Comment (lemunozm) [inline include/xtypes/Assert.hpp:60]: Cool!!

File: include/xtypes/EnumeratedType.hpp [hunk 4]
```diff
@@ -44,11 +43,13 @@ class EnumeratedType : public PrimitiveType<T>
 
     /// \brief Get a enumerator value by name. O(log(n)).
     /// \param[in] name EnumMember name.
-    /// \pre has_member() == true
+    /// \pre has_enumerator() == true
```
Comment (lemunozm) [inline include/xtypes/EnumeratedType.hpp:46]: good view

**Review by lemunozm — CHANGES_REQUESTED** [review]

File: include/xtypes/Assert.hpp (hunk 1 above)
Comment (LuisGP) [reply include/xtypes/Assert.hpp]: Do you see any advantage in this case?

File: include/xtypes/Assert.hpp (hunk 3 above)
Comment (LuisGP) [reply include/xtypes/Assert.hpp:60]: Stackoverflow ™

File: include/xtypes/Assert.hpp (hunk 1 above)
Comment (lemunozm) [reply include/xtypes/Assert.hpp]: I think that stringstream works better with the string concatenation (but I am thinking now that could be equals). Anyway, the stringstream avoid you to use the std:string for each append.

Comment (LuisGP) [reply include/xtypes/Assert.hpp]: Once modified, it improves readability a lot. Nice!

**Review by lemunozm — APPROVED** [review]

**Pull request merged**